Editor/MenuActions/Editors/NewPolyShape.cs

```diff
@@ -34,6 +34,23 @@ public override bool enabled
 
         public override ActionResult DoAction()
         {
+            var inspWindow = EditorWindow.GetWindow<InspectorWindow>();
+            if (inspWindow != null && inspWindow.isLocked == true)
```

Comment:
This is assuming there is only a single Inspector window open. There may be multiple, with any variation of locked and unlocked.

Reply:
I will look at all Inspector window. This will be only an issue if everything is locked.